spec/support/matchers/have_includes.rb

```diff
@@ -0,0 +1,32 @@
+RSpec::Matchers.define :have_includes do |*includes|
```

Comment:
This matcher somewhat ballooned in complexity and is making me rethink our decision around this 🤔 As this is all untested code really, maybe if we go back to something simple as this, maybe it's enough for now:
```
expect(result[:included].map { |r| r[:type] }).to contain_exactly(%w[people moves locations locations])
```
This ties into a bigger conversation around comparing nested JSON, and although `include_json` does resolve some of it I think there is something missing which we can add. Let me have a ponder and do a bit of research around that, but I guess it's outside the scope of this pr for now. Let me know your thoughts

Reply:
I think that given I am specifying the exact ids and entities that are being included I don't need to validate the types? And actually am implicitly validating them?

Reply:
```
            from_locations = allocations.map(&:from_location)
            to_locations = allocations.map(&:to_location)
            moves = allocations.flat_map(&:moves)
            people = moves.map { |move| move&.profile&.person }
```

This bit ^^

Reply:
This is also super visible in the expectation so is really obvious what we're testing around the types of things that are getting returned.